File: client-sdk/csharp/sdk/src/VertexCache/Sdk/Protocol/Parsers/PongResponseParser.cs
using VertexCache.Sdk.Results;

namespace VertexCache.Sdk.Protocol.Parsers
{
    public class PongResponseParser : IResponseParser
    {
        public bool CanParse(string response) => response == "+PONG";

        public VCacheResult Parse(string response) => VCacheResult.SuccessWithRaw(response);
    }
}

File: client-sdk/go/sdk/protocol/parsers/pong_response_parser.go
package parsers

import "vertexcache/sdk/results"

type PongResponseParser struct{}

func (p *PongResponseParser) CanParse(response string) bool {
	return response == "+PONG"
}

func (p *PongResponseParser) Parse(response string) *results.VCacheResult {
	return results.NewSuccessWithRaw(response)
}
